plugins/trace_api_plugin/include/eosio/trace_api/trace.hpp

```diff
@@ -36,6 +47,17 @@ namespace eosio { namespace trace_api {
      chain::transaction_header                  trx_header = {};
   };
 
+struct transaction_trace_v2 {
```

Comment:
This is a lot of duplication for both this and the block_trace.

Instead, why not refactor out the `action_trace_v1/v2` parameter.  This is only going to be exacerbated as we add more action_trace types.  That way you only have the two definitions, "v0" and "v1" and then some using statements to give it which action_trace type for which version we want.

``` c++
template <typename ActionTrace>
struct transaction_trace_base {
   chain::transaction_id_type id = {};
   std::vector<ActionTrace> actions = {};
};
```

Reply:
good idea.